libsolidity/ast/AST.cpp

```diff
@@ -480,6 +480,66 @@ bool VariableDeclaration::canHaveAutoType() const
 	return isLocalVariable() && !isCallableParameter();
 }
 
+bool VariableDeclaration::isEventParameter() const
+{
+	return dynamic_cast<EventDefinition const*>(scope()) != nullptr;
+}
+
+bool VariableDeclaration::isReferenceType() const
+{
+	return dynamic_cast<ReferenceType const*>(typeName()->annotation().type.get());
+}
+
+set<VariableDeclaration::Location> VariableDeclaration::getAllowedStorageLocations() const
```

Comment:
We try to avoid the `get` prefix if it is obvious, so please rename to `allowedStorageLocations()`